### ISSUE TITLE
Dataflow doesn't respect the project part of subnetworkRef.external

### ISSUE THREAD

**Issue by deverant** [GoogleCloudPlatform/k8s-config-connector#255]
**Describe the bug**
Streaming dataflow job tries to use a subnetwork that is combination of the `subnetworkRef.external` field and the project where the dataflow job is being ran. Expected behavior is that the dataflow job would use the `subnetworkRef.external` field as-is.

Example `subnetworkRef.external` value:
```
subnetworkRef:
  external: https://www.googleapis.com/compute/v1/projects/PROJECT_A/regions/europe-west1/subnetworks/SUBNETWORK
```

Example of the error when the dataflow job fails to start:
```
Workflow failed. Causes: Subnetwork https://www.googleapis.com/compute/v1/projects/PROJECT_B/regions/europe-west1/subnetworks/SUBNETWORK is not accessible to Dataflow Service account or does not exist
```

Notice that the project has changed between the `subnetworkRef.external` and the error message. PROJECT_B in this case refers to the project where the dataflow job is being executed. In our case PROJECT_B is also the namespace that is being used (we map namespaces one-to-one to projects).

**ConfigConnector Version**
1.17.0

**To Reproduce**
*Steps to reproduce the behavior:*
1. Setup a subnetwork in one project
2. Try to setup a dataflow job in another project that uses that 

*YAML snippets:*

```yaml
apiVersion: dataflow.cnrm.cloud.google.com/v1beta1
kind: DataflowJob
metadata:
  annotations:
    cnrm.cloud.google.com/on-delete: "cancel"
  name: example-job
  namespace: PROJECT_B
spec:
  tempGcsLocation: gs://example/dataflow-temp
  templateGcsPath: gs://example/dataflow-templates/TEMPLATE_NAME
  parameters: []
  zone: europe-west1-d
  machineType: n1-standard-2
  maxWorkers: 1
  serviceAccountRef:
    name: service-account
  subnetworkRef:
    external: https://www.googleapis.com/compute/v1/projects/PROJECT_A/regions/europe-west1/subnetworks/SUBNETWORK
```

**Comment by caieo** [comment]
Hi @deverant, thank you for reporting this bug. External references is definitely not working the way it should be here and we will look into it. Can you shed some light on how much this bug is impacting you and how urgently you need this fixed?

**Comment by jcanseco** [comment]
Hi @deverant, this has been fixed in KCC 1.19.1.

**Comment by deverant** [comment]
I can confirm this issue was fixed for me in 1.19.1.
